pydicom/dataset.py

```diff
@@ -1808,12 +1811,35 @@ def __setattr__(self, name, value):
                              'element and must be added using '
                              'the add() or add_new() methods.'
                              .format(name))
+        elif name == "file_meta":
+            self._set_file_meta(value)
         else:
             # name not in dicom dictionary - setting a non-dicom instance
             # attribute
             # XXX note if user mis-spells a dicom data_element - no error!!!
             object.__setattr__(self, name, value)
 
+    def _set_file_meta(self, value):
+        if value is not None:
```

Comment:
Maybe leave that check to the `FileMetaDataset` itself? If it already is a `FileMetaDataset`, it cannot have invalid tags, and if it isn't, you could warn and convert it to one, which will also do the check.

Reply:
We can't convert it and assign to `ds.file_meta`, because we would lose the connection to the user's `Dataset` instance.  I did try creating it and not using it, to let it run its own checks like you said, but it seemed wasteful to instantiate the entire object.  Perhaps could make a  `non_group_2` method or `validate_file_meta`-- maybe a static class method of `FileMetaDataset`?

Reply:
Ah, you are right - hadn't thought of this. Yes, maybe factor the check out into an own method...